### ISSUE TITLE
Add `LNLSTMCell`: current API does not support combining `nn.LayerNorm` and LSTMCell

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#12167]
[Current implementation](https://github.com/pytorch/pytorch/blob/04c0971679374148aa4105e2e998de6478eca1eb/torch/nn/modules/rnn.py#L723-L727) of LSTMCell calls internal C function which makes it infeasible to write a layer normalization of LSTM cell by wrapping it with `nn.LayerNorm`. 

It would be great to add this feature. 

We could add trainable parameters inside `RNNCellBase` as follows
```python
class RNNCellBase(nn.Module):
    def __init__(self, input_size, hidden_size, bias, num_chunks):
        super(RNNCellBase, self).__init__()
        self.input_size = input_size
        self.hidden_size = hidden_size
        self.bias = bias
        self.weight_ih = Parameter(torch.Tensor(num_chunks * hidden_size, input_size))
        self.weight_hh = Parameter(torch.Tensor(num_chunks * hidden_size, hidden_size))
        self.ln_ih = nn.LayerNorm(num_chunks * hidden_size)
        self.ln_hh = nn.LayerNorm(num_chunks * hidden_size)
        self.ln_c = nn.LayerNorm(hidden_size)
```
However, the it is impossible to use it for the computation 
```python
return _VF.lstm_cell(
            input, hx,
            self.weight_ih, self.weight_hh,
            self.bias_ih, self.bias_hh,
        )
```
as it calls C functions. 

And writing it purely in Python, the speed would be too slow.

**Comment by weiyangfb** [comment]
@zou3519

**Comment by SsnL** [comment]
Duplicate of #12482
